mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -0,0 +1,131 @@
+"""Class for monitoring system stats."""
+
+import logging
+import threading
+
+from mlflow.environment_variables import (
+    MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING,
+    MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL,
+)
+from mlflow.system_metrics.metrics.cpu_monitor import CPUMonitor
+from mlflow.system_metrics.metrics.gpu_monitor import GPUMonitor
+
+_logger = logging.getLogger(__name__)
+
+
+class SystemMetricsMonitor:
+    """Class for monitoring system stats.
+
+    This class is used for pulling system metrics and logging them to MLflow. Calling `start()` will
+    spawn a thread that logs system metrics periodically. Calling `finish()` will stop the thread.
+    Logging is done on a different frequency from pulling metrics, so that the metrics are
+    aggregated over the period. Users can change the logging frequency by setting
+    `$MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL` and `$MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING`
+    environment variables.
+
+    Args:
+        mlflow_run: an 'mlflow.entities.run.Run' instance, which is used to bootstrap system metrics
+            logging with the MLflow tracking server.
```

Comment:
I think this should be a string `run_id`, rather than an MLflow Run entity. This way, users can construct and run it directly (see https://github.com/mlflow/mlflow/pull/9557/files#r1323692384)

Reply:
I don't have strong opinion on this, I was choosing `run` because users can just do `monitor = SystemMetricsMonitor(run)`, but since this is not our critical user journey, either is fine to me.

Changed to `run_id`.